flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/AlterSchemaConverter.java

```diff
@@ -916,15 +987,29 @@ private Operation buildAlterTableChangeOperation(
             List<TableChange> tableChanges,
             Schema newSchema,
             ResolvedCatalogTable oldTable) {
+        CatalogTable.Builder builder =
+                CatalogTable.newBuilder()
+                        .schema(newSchema)
+                        .comment(oldTable.getComment())
+                        .partitionKeys(oldTable.getPartitionKeys())
+                        .options(oldTable.getOptions());
+
+        if (alterTable instanceof SqlAlterTableSchema) {
```

Comment:
why can't we simply call `oldTable.getDistribution`?

Reply:
Ah, I ended up adding `oldTable` towards the end of my work.  I can try to use it here.

Reply:
After looking at this again, I figured out that we need to create the new table.  Since that table needs the added/modified distribution, we have to take it from the `SqlAlterTableSchema` rather than the `oldTable`.